packages/react/src/table/TableScroller.tsx

```diff
@@ -204,140 +215,212 @@ export function TableScroller({ children }: TableScrollerProps) {
-							display: 'none',
-						},
 				}}
-				focusRingFor="keyboard"
-				onScroll={repositionThumb}
-				ref={scrollerRef}
-				tabIndex={0}
-			>
-				{children}
-			</Box>
-			<Flex
-				background="body"
-				alignItems="center"
-				css={{
-					bottom: 0,
-					display: thumbWidthRatio === 1 ? 'none' : undefined,
-					left: 0,
-					position: 'sticky',
-					right: 0,
-				}}
-				flexWrap="nowrap"
-				gap={0.25}
 			>
 				<Box
-					aria-hidden
-					as="button"
+					data-table-left-overlay="true"
+					css={{
+						left: overlayOffsets?.left,
+						pointerEvents: 'none',
+						position: 'fixed',
+						zIndex: SCROLL_OVERLAY_Z_INDEX,
+						...(scrollerHeight && { height: `${scrollerHeight / 16}rem` }),
+
+						// Switch between single line at full scroll and overlay shadow at partial scroll
+						...(scrollerRef.current?.scrollLeft === 0
+							? {
+									background: boxPalette.borderMuted,
+									width: '1px',
+							  }
+							: {
+									background: `linear-gradient(to right, rgba(0, 0, 0, 0.20), transparent)`,
+									width: '1rem',
+							  }),
+
+						// Container queries not working with dynamic styles, so we'll hide the overlay up until the breakpoint instead
+						[maxContainerBreakpointForFrozenColumns]: {
+							'[data-table-left-overlay="true"]': {
+								left: 0,
+							},
+						},
+					}}
+				/>
+				<Box
 					css={{
-						appearance: 'none',
-						background: 'none',
-						cursor: 'default',
-						height: pxToRem(24),
-						width: pxToRem(24),
+						msOverflowStyle: 'none',
+						overflowX: 'auto',
+						overscrollBehaviorX: 'none',
+						scrollbarWidth: 'none',
+						WebkitOverflowScrolling: 'touch',
+						width: '100%',
+						'&::-webkit-scrollbar, &::-webkit-scrollbar-thumb, &::-webkit-scrollbar-track':
+							{
+								display: 'none',
+							},
 					}}
-					onClick={() => handleButtonClick('left')}
-					onMouseDown={(event: React.MouseEvent) =>
-						handleButtonPress(event, 'left')
-					}
-					onMouseUp={handleButtonRelease}
-					onMouseLeave={handleButtonRelease}
-					onTouchStart={(event: React.TouchEvent) =>
-						handleButtonPress(event, 'left')
-					}
-					onTouchEnd={handleButtonRelease}
-					tabIndex={-1}
-					type="button"
+					focusRingFor="keyboard"
+					onScroll={repositionThumb}
+					tabIndex={0}
+					ref={scrollerRef}
 				>
-					<ScrollbarArrowLeftIcon color="border" />
+					{children}
 				</Box>
 				<Box
-					aria-hidden
-					background="shade"
-					border
-					borderColor="muted"
+					data-table-right-overlay="true"
+					css={{
+						pointerEvents: 'none',
+						position: 'fixed',
+						right: overlayOffsets?.right,
+						zIndex: SCROLL_OVERLAY_Z_INDEX,
+						...(scrollerHeight && { height: `${scrollerHeight / 16}rem` }),
+
+						// Switch between single line at full scroll and overlay shadow at partial scroll
+						...(isScrolledToEnd
+							? {
+									background: boxPalette.borderMuted,
+									width: '1px',
```

Comment:
As above, I don't think this should be applied if the column isn't frozen 
<img width="54" alt="image" src="https://github.com/user-attachments/assets/0562e2ef-ded5-4940-b6e6-22bed632da97">